### PR TITLE
[on-chain config] add fetch_config method to OnChainConfig trait

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

Adding `fetch_config` method to reduce repeated code in loading VM config

### CODE REVIEW

File: types/src/on_chain_config.rs [hunk 1]
```diff
@@ -78,6 +78,15 @@ pub trait OnChainConfig: Send + Sync + DeserializeOwned {
...
         Self::deserialize_default_impl(bytes)
     }
+
+    fn fetch_config<T>(storage: T) -> Option<Self>
+    where
+        T: ConfigStorage,
+    {
+        storage
+            .fetch_config(Self::CONFIG_ID.access_path())
+            .and_then(|bytes| Self::deserialize_into_config(&bytes).ok())
```
Comment (runtian-zhou) [inline types/src/on_chain_config.rs:88]: Sorry I removed the `error!` log in my previous PR. Should we add it back here?

Comment (christinacelee) [reply types/src/on_chain_config.rs:88]: we can add logging at the function callsite for better context (vs inside this function) - can do it in this PR

**Pull request closed without merging**